src/ray/gcs/client.cc

```diff
@@ -37,8 +104,41 @@ AsyncGcsClient::AsyncGcsClient(CommandType command_type)
 AsyncGcsClient::AsyncGcsClient() : AsyncGcsClient(ClientID::from_random()) {}
 
 Status AsyncGcsClient::Connect(const std::string &address, int port, bool sharding) {
-  RAY_RETURN_NOT_OK(context_->Connect(address, port, sharding));
-  RAY_RETURN_NOT_OK(primary_context_->Connect(address, port, /*sharding=*/false));
+  RAY_RETURN_NOT_OK(primary_context_->Connect(address, port));
+  std::vector<std::string> addresses{address};
+  std::vector<int> ports{port};
```

Comment:
Why do we initialize `addresses` and `ports` with the `address` and `port` of the primary shard? Don't we want them to start out empty and then contain just the shard addresses/ports?

Reply:
Yep. I mixed up the "no-sharding scenario", where all of those share the primary shard, and "sharding scenario", where they use separate shards here.